File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge38_test.rb
# frozen_string_literal: true

require_relative './Challenge38'

TESTS = { input: %w[00010101 1111 0000 10100101 00000001],
          output: [21, 15, 0, 165, 1] }.freeze

TESTS[:input].each_with_index do |test, index|
  puts "\norigina: #{test}"
  puts binary_to_decimal(test)
  puts "expected: #{TESTS[:output][index]}"
end

File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge38_test.js
const binaryToDecimal = require("./Challenge38.js");

const tests = {
  input: ["00010101", "1111", "0000", "10100101", "00000001"],
  output: [21, 15, 0, 165, 1],
};

tests.input.forEach((test, index) => {
  console.log("\noriginal:", test);
  console.log(binaryToDecimal.func(test));
  console.log("expected:", tests.output[index]);
});
